File: localstorage/src/main/java/ch/fhnw/geiger/localstorage/db/mapper/AbstractMapper.java
package ch.fhnw.geiger.localstorage.db.mapper;

import ch.fhnw.geiger.localstorage.StorageException;
import ch.fhnw.geiger.localstorage.db.GenericController;
import ch.fhnw.geiger.localstorage.db.StorageMapper;
import ch.fhnw.geiger.localstorage.db.data.Node;

/**
 * <p>An abstract mapper providing general checks.</p>
 */
public abstract class AbstractMapper implements StorageMapper {

  protected void getSanity(String path) throws StorageException {
    if (path == null) {
      throw new NullPointerException("path may not be null");
    }
  }

  protected void checkPath(Node node) throws StorageException {
    checkPath(node.getPath());
  }

  protected void checkPath(String path) throws StorageException {
    if (path == null) {
      throw new StorageException("illegal path (may not be null)");
    }
    if (":".equals(path)) {
      return;
    }
    if (!path.matches("(" + GenericController.PATH_DELIMITER + "[a-zA-Z0-9\\-]+)*")) {
      throw new StorageException("illegal path detected in \"" + path + "\"");
    }
  }

}

File: lib/src/db/mapper/abstract_mapper.dart
library geiger_localstorage;

import 'package:geiger_localstorage/src/storage_exception.dart';

import '../generic_controller.dart';
import '../storage_mapper.dart';

/// <p>An abstract mapper providing general checks.</p>
abstract class AbstractMapper with StorageMapper {
  void getSanity(String? path) {
    if (path == null) {
      throw NullThrownError();
    }
  }

  void checkPath(String path) {
    if (':' == path) {
      return;
    }
    if (!RegExp('(' + GenericController.pathDelimiter + '[a-zA-Z0-9\\-]+)*')
        .hasMatch(path)) {
      throw StorageException(('illegal path detected in "' + path) + '"');
    }
  }
}
